yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
                 "could not determine target color type "
                 "from target='{}' to '{}'"
             ).format(self.target_type, self._target_color_type))
+
+
+##########################################################################
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
+        if proj_dim not in [2, 3]:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.proj_dim = proj_dim
+        self.alpha = alpha
+        self._cax = None
+
+    @property
+    def cax(self):
```

Comment:
So we may have to think in more detail about how to best organize all of the related axes in the visualizer. I do like what you've done here, which is to specify the colorbar axes as a property but raise an exception if the visualizer does not have a colorbar. The question is do we do this for all possible axes in the visualizer, or do we leave it for the layout mechanism of the subclasses?

All projection visualizers will have:

1. A scatter plot 
2. An optional colorbar in the case of `_target_type == "CONTINUOUS"`
3. In the 3D case, an Axes3D object that replaces gca 

PCA will additionally have the heatmap, JointPlot will have the X and Y histograms. 

Perhaps we should have a `layout()` method that is called by `draw()` and can be overridden by subclasses? I like that you've added `layout_colorbar` - but this would make things a bit more general.

Reply:
I just need to change the name to `layout`, right? Just to be sure, in PCA we will have two colorbar and a heatmap for the continuous case, right?

Reply:
Yep, changing the name to `layout` is a good start! Why two colorbars in PCA? One for the data points and one for the heatmap?

Reply:
Yup!